src/picongpu/include/plugins/adios/restart/ReadAttribute.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2016 Alexander Grund
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "simulation_types.hpp"
+#include "plugins/adios/ADIOSWriter.def"
+#include "debug/VerboseLog.hpp"
+#include "traits/PICToAdios.hpp"
+#include <adios.h>
+#include <adios_read.h>
+#include <adios_error.h>
+#include <stdexcept>
+
+namespace picongpu {
+namespace adios {
+
+    /**
+     * Read an attribute from an open ADIOS file, check that its type is correct and return it
+     *
+     * @param fp       Open ADIOS file handle
+     * @param basePath Path where the attribute is located in the file (with or w/o trailing slash)
+     * @param attrName Name of the attribute. Used for status output and concatenated with basePath
+     * @retval Attribute value
+     */
+    template<typename T_Attribute>
+    T_Attribute readAttribute(ADIOS_FILE* fp, const std::string& basePath, const std::string& attrName)
```

Comment:
don't we just the the same thing while reading `sim_slides` but way less verbose?